src/cace/ari/text_str.yac

```diff
@@ -262,7 +263,18 @@ ssp-literal:
             }
             else
             {
-                //FIXME where to convert (and check) type?
+              ari_t tmp;
+              ari_init(&tmp);
+              int err = amm_type_convert(amm_type_get_builtin(ARI_TYPE_TP), &tmp, found_r);
+              if (err) 
+              {
+                ari_deinit(found_r);
+                found_r = NULL;
+              }
+              else
+              {
+                ari_set_copy(found_r, &tmp);
```

Comment:
This can be `ari_set_move()` to avoid needing to deinit the `tmp` var.

Reply:
I changed the call, however I noticed the code in `ari_set_move` is:

```
// reset the state of the src (not deinit)
ari_reset(src);
```

It may not matter in this case, but in general it would seem `ari_deinit` still needs to be called after `ari_set_move`?

Reply:
This is a good point the deinit is still needed after a move from. Is the doxygen comment for move funce clear about this? Might need an update if not.

Reply:
Updated doxygen comments and added deinit calls.